source/rock/middle/Addon.ooc

```diff
@@ -158,6 +159,10 @@ Addon: class extends Node {
                 if (fDecl hasThis() && !call getExpr()) {
                     // add `this` if needed.
                     call setExpr(VariableAccess new("this", call token))
+                    /* Not tested
+                    if (res params explicitThis) {
+                        call token printMessage("Implicit this detected in...")
+                    }*/
```

Comment:
I could probably use `fDecl` to get the name of the addon function but we don't have any real world use case to confirm that there are no special cases where we should not warn. Using addon would make it very hard to find the function being called which is already difficult with overloading and many layers of inheritance.